### PR TITLE
Add verbose log level for filenames being bundled

### LINKED ISSUE #363: Deploy Should Optionally List Packaged Files
I just discovered that rsconnect-deploy was uploading several files I did not want to be shared. Could there be an additional flag that would list every file as it is being packaged for deployment? Perhaps an additional verbosity level (`-vv`)?

Furthermore, it would be nice if there were a dry-run like configuration which would list everything that is to be uploaded without doing anything.

### PR DESCRIPTION
**Pull request by mmarchetti**
This PR adds a new log level, and emits messages at that level for each file that is added to a bundle. The new level is used when `-v` is present on the command line; use `-vv` for full debug logging.

## Intent
Fixes #363 

## Type of Change

- [ ] Bug Fix            
- [x] New Feature       
- [ ] Breaking Change   

## Approach
Introduced a custom log level `VERBOSE` between `INFO` and `DEBUG`, and configure log level appropriately based on the verbosity. Change verbosity from a bool/flag to a counter/int.

## Automated Tests
Our tests don't validate logged messages.

## Directions for Reviewers
Deploy various content types (notebook, app/API, and manifest) with no verbosity, `-v`, and `-vv`. With `-v`, and `-vv`, you should see messages logged at level `VERBOSE` showing the files added to the bundle.

## Checklist

- [x] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.
- [x] I have updated all related GitHub issues to reflect their current state.

### CODE REVIEW

File: rsconnect/main.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def server_args(func):
         type=click.Path(exists=True, file_okay=True, dir_okay=False),
         help="The path to trusted TLS CA certificates.",
     )
-    @click.option("--verbose", "-v", is_flag=True, help="Print detailed messages.")
+    @click.option("--verbose", "-v", count=True, help="Enable verbose output. Use -vv for very verbose (debug) output.")
```
Comment (tdstein) [inline rsconnect/main.py:138]: Nice.

**Review by tdstein — COMMENTED** [review]
LGTM.

I see copious amounts of logging output when setting `-vv`.  Less on `-v`. And normal logging by default.

minor question; afaik `VERBOSE` isn't a standard log level. Should we be utilizing `TRACE` somehow?

**Comment by mmarchetti** [comment]
> minor question; afaik `VERBOSE` isn't a standard log level. Should we be utilizing `TRACE` somehow?

I thought about that, but it didn't feel like trace output?

**Review by kgartland-rstudio — APPROVED** [review]
Validated. Deployed each content type + manifests with `-v` and `-vv`. All content deployed and shows appropriate log level ([VERBOSE] and [DEBUG]).

**Pull request merged**